frame/staking/src/pallet/impls.rs

```diff
@@ -744,22 +745,18 @@ impl<T: Config> Pallet<T> {
 	/// Get the targets for an upcoming npos election.
 	///
 	/// This function is self-weighing as [`DispatchClass::Mandatory`].
-	pub fn get_npos_targets() -> Vec<T::AccountId> {
-		let mut validator_count = 0u32;
-		let targets = Validators::<T>::iter()
-			.map(|(v, _)| {
-				validator_count.saturating_inc();
-				v
-			})
+	pub fn get_npos_targets(maybe_max_len: Option<usize>) -> Vec<T::AccountId> {
+		let targets = T::TargetList::iter()
+			.take(maybe_max_len.unwrap_or_else(Bounded::max_value))
```

Comment:
`Bounded::max_value` is usize::MAX? This is a bit scary, especially since the point of using the bags list here is to allow it to be virtually unbounded. IMO we should either hardcode a fallback or have a configurable fallback value.